test/include/problems/FixedPointProblem.h

```diff
@@ -0,0 +1,39 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef FixedPointProblem_H
+#define FixedPointProblem_H
+
+#include "FEProblem.h"
+
+class FixedPointProblem;
+
+template <>
+InputParameters validParams<FixedPointProblem>();
+
+/**
+ * FEProblem derived class for lagging a tagged residual.
+ */
+class FixedPointProblem : public FEProblem
```

Comment:
Yes to the first. No to the second. We cannot derive this from `FEProblemBase` because both` FEProblem` and `EigenProblem` are not going to have the code in this `FixedPointProblem`. To do that, we will have to derive `FEProblem` and `EigenProblem` separately. We can use an interface to remove the code duplication. But it could be desired to merge `EigenProblem` into `FEProblem` if possible.

Reply:
This is actually quite an issue to me: @fdkong can we merge `EigenProblem` into `FEProblem`?